### PR TITLE
feat: fix rest of warnings for accessibility

### PR DESCRIPTION
**Pull request by mavmaf**
## Motivation and context
All accessibility warnings introduced by the eslint accessibility plugin are fixed now. Therefore, from now on, any piece of code not complying these rules will through an error.
Also literal have been extracted from the code and set as attributes.

- [ ] Dependencies. If any, specify:
- [ ] Open issue. If applicable, link:

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

You can remove the changes in `.vue` files while keeping the new `.eslintrc.js` file and analyse the errors and how they were solved.

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/components/base-dropdown.vue [hunk 1]
```diff
@@ -34,6 +34,7 @@
         :aria-expanded="isOpen.toString()"
         class="x-dropdown__items-list"
         role="listbox"
+        tabIndex="-1"
```
Comment (tiborux) [inline packages/x-components/src/components/base-dropdown.vue:37]: Should be 0? 

[TabIndex](https://developer.mozilla.org/es/docs/Web/HTML/Global_attributes/tabindex)

File: packages/x-components/src/views/home/Home.vue [hunk 2]
```diff
@@ -154,7 +162,7 @@
               </template>
             </SortDropdown>
 
-            <RenderlessExtraParams #default="{ value, updateValue }" name="store">
+            <RenderlessExtraParams v-slot="{ value, updateValue }" name="store">
```
Comment (mnavarroespinosa) [inline packages/x-components/src/views/home/Home.vue:165]: I guess it was auto-fixed by eslint, we should keep with `#default`.
```suggestion
            <RenderlessExtraParams #default="{ value, updateValue }" name="store">
```

File: packages/x-components/src/views/home/Home.vue [hunk 3]
```diff
@@ -260,7 +268,7 @@
                         :max="controls.slicedFilters.max"
                         :data-test="`${facet.label}-sliced-filters`"
                       >
-                        <SelectedFilters #default="{ selectedFilters }" :facetsIds="[facet.id]">
+                        <SelectedFilters v-slot="{ selectedFilters }" :facetsIds="[facet.id]">
```
Comment (mnavarroespinosa) [inline packages/x-components/src/views/home/Home.vue:271]: I guess it was auto-fixed by eslint, we should keep with `#default`.
```suggestion
                        <SelectedFilters #default="{ selectedFilters }" :facetsIds="[facet.id]">
```

File: packages/x-components/src/views/home/Home.vue [hunk 4]
```diff
@@ -294,7 +302,7 @@
         <template #main-body>
           <!--  Redirection  -->
           <Redirection
-            #default="{ redirection, redirect, abortRedirect, isRedirecting, delayInSeconds }"
+            v-slot="{ redirection, redirect, abortRedirect, isRedirecting, delayInSeconds }"
```
Comment (mnavarroespinosa) [inline packages/x-components/src/views/home/Home.vue:305]: I guess it was auto-fixed by eslint, we should keep with `#default`.
```suggestion
            #default="{ redirection, redirect, abortRedirect, isRedirecting, delayInSeconds }"
```

File: packages/x-components/src/views/home/Home.vue [hunk 5]
```diff
@@ -363,7 +371,7 @@
                               :max-items-to-render="3"
                             >
                               <NextQuery
-                                #default="{ suggestion: nextQuery }"
+                                v-slot="{ suggestion: nextQuery }"
```
Comment (mnavarroespinosa) [inline packages/x-components/src/views/home/Home.vue:374]: I guess it was auto-fixed by eslint, we should keep with `#default`.
```suggestion
                                #default="{ suggestion: nextQuery }"
```

File: packages/x-components/src/views/home/Home.vue [hunk 6]
```diff
@@ -402,7 +410,7 @@
             <!-- Recommendations -->
             <Recommendations v-if="!$x.query.search || $x.noResults" #layout="{ recommendations }">
               <BaseVariableColumnGrid
-                #default="{ item: result }"
+                v-slot="{ item: result }"
```
Comment (mnavarroespinosa) [inline packages/x-components/src/views/home/Home.vue:413]: I guess it was auto-fixed by eslint, we should keep with `#default`.
```suggestion
                #default="{ item: result }"
```

File: packages/x-components/src/views/home/Home.vue [hunk 7]
```diff
@@ -126,7 +134,7 @@
           >
             <span>{{ $x.totalResults }} Results</span>
             <BaseColumnPickerList
-              #default="{ column }"
+              v-slot="{ column }"
```
Comment (mnavarroespinosa) [inline packages/x-components/src/views/home/Home.vue:137]: I guess it was auto-fixed by eslint, we should keep with `#default`.
```suggestion
              #default="{ column }"
```

File: packages/x-components/src/views/home/Home.vue (hunk 2 above)
Comment (mavmaf) [reply packages/x-components/src/views/home/Home.vue:165]: good catch, thanks!

**Review by mnavarroespinosa — APPROVED** [review]
Looks good to me!

**Review by tiborux — APPROVED** [review]

**Pull request merged**